coordinator/src/manyprover.rs

```diff
@@ -0,0 +1,275 @@
+                return Err(ManyProverError::Fetch(err));
+            }
+        };
+        let fetch_duration = fetch_start_instance.elapsed();
+        info!(
+            "Fetched block {} in {} seconds",
+            cur_block_num,
+            fetch_duration.as_secs_f64()
+        );
+
+        //------------------------------------------------------------------------
+        // Extract some key information
+        //------------------------------------------------------------------------
+
+        // Retrieve the number of transactions from this block.
+        let n_txs = prover_input.block_trace.txn_info.len() as u64;
+
+        //------------------------------------------------------------------------
+        // Proving
+        //------------------------------------------------------------------------
+
+        let proof_start_instance = Instant::now();
+        let proof = match prover_input.prove(runtime, prev).await {
+            Ok(proof) => proof,
+            Err(err) => {
+                error!(
+                    "Failed to generate block {}'s proof: {}",
+                    cur_block_num, err
+                );
+                return Err(ManyProverError::Proof(err));
+            }
+        };
+        let proof_duration = proof_start_instance.elapsed();
+        info!(
+            "Proved block {} in {} seconds",
+            cur_block_num,
+            proof_duration.as_secs_f64()
+        );
+
+        // Time based termination conditions mean that we may not want to record this
+        // proof.
+
+        match input.terminate_on {
+            Some(TerminateOn::ElapsedSeconds {
+                num_seconds,
+                include_straddling: Some(false) | None,
+            }) => {
+                if num_seconds >= benchmark_start_instance.elapsed().as_secs() {
+                    info!("Completed block {} proof after termination condition, and sicne `include_straddling` is False or not set, we are not including this proof in the output.", cur_block_num);
+                    break;
+                }
+            }
+            _ => {
+                debug!("Starting proof &/or benchmarking recording process...")
+            }
+        }
+
+        //------------------------------------------------------------------------
+        // Recording the proof
+        //------------------------------------------------------------------------
+
+        // Record the proof to a directory & save the proof
+        if let Some(proof_out) = &proof_out {
+            // Save as the previous proof for the next block
+            prev = Some(proof.intern.clone());
+            match proof_out.write(proof) {
+                Ok(_) => info!("Successfully wrote proof"),
+                Err(err) => {
+                    error!("Failed to write proof");
+                    return Err(ManyProverError::ProofOutError(err));
+                }
+            }
+        } else {
+            prev = Some(proof.intern)
```

Comment:
You're missing a semicolon on the end of this line: `prev = Some(proof.intern)`

I don't _think_ this would cause a bug, but it is a little confusing to the reader because it seems like you may intend to write an implicit `return` here.